crates/database/interface/src/lib.rs

```diff
@@ -244,7 +307,9 @@ pub trait DatabaseCommitExt: Database + DatabaseCommit {
             // Unfortunately must collect here to short circuit on error
             .collect::<Result<Vec<_>, _>>()?;
 
-        self.commit_iter(transitions);
+        // Use commit() instead of commit_iter() since we already have a collected Vec.
+        // This keeps these methods dyn-compatible without requiring Self: Sized.
+        self.commit(transitions.into_iter().collect());
```

Comment:
This is going to be a perf penalty as it now recreates the HashMap. Should we enforce Sized here too?

Reply:
Have dug litttle bit deeper on this, adding Sized basically removed this function from `Box<dyn Trait>`. Adding `&mut dyn IntoIterator` would allowed for trait to be boxable but still to have IntoIter function. Assumption here is that commit_iter would handle a lot of accesses to hashmaps, so the impact of dynamic dispatch of dyn would not be noticeable.

There is an additional nice small trick where you can do, this should make api litle nicer, but didn't try to to be sure.
```
impl dyn DatabaseCommit {
    pub fn temp(&mut self, iter: impl IntoIterator<..>) {}
}